src/Compilers/CSharp/Test/Semantic/Semantics/FunctionPointerTests.cs

```diff
@@ -1651,7 +1754,6 @@ public static void M()
     }
 }");
 
-            // These should all work: https://github.com/dotnet/roslyn/issues/39865
```

Comment:
These could only be inferred if we supported array covariance for implicit pointer conversions. The spec makes no mention of such support, and we do not support `int*[]` being converted to `void*[]` today, so this is not expected to be supported.